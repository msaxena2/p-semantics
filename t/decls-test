in "../builtins.maude"
in "../p-syntax.maude"

mod DECLS-TEST is
  including P-SYNTAX .

endm

parse var x ; .
parse var x ; var y ; .
parse var x : Int ; .
parse var x : Int ; var y : Int ; .

parse
  state foo {
    var x ; var y : Int ;
  } .

parse
  on eFoo do {
    var x ;
  } .

parse
  on eFoo do {
    var x ;
    goto foo ;
  } .

parse 1 + a .

parse 1 + a + 1 .

parse x = a + 1 ; .

parse if (a <= 1)  x = 1 ; else x = 2 ; .

parse if (a <= 1) { x = 1 ; y = 2 ; } else  x = 2 ;  .

quit .
