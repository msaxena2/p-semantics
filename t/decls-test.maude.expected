Stmt: var x ;
Stmt: var x ;
var y ;
Stmt: var x : Int ;
Stmt: var x : Int ;
var y : Int ;
Stmt: state foo {
  var x ;
  var y : Int ;
}
Stmt: on eFoo do {
  var x ;
}
Stmt: on eFoo do {
  var x ;
  goto foo ;
}
Exp: 1 + a
Exp: 1 + a + 1
Stmt: x = a + 1 ;
Exps: a : Int,b : Int
Exps: a : Int,b : Int
Stmt: if (a <= 1) x = 1 ;
else x = 2 ;
Stmt: if (a <= 1) {
  x = 1 ;
  y = 2 ;
}
else x = 2 ;
Stmt: machine foo ; {
  var x : Int ;
  var y : Int ;
  on eBar do {
    var z ;
    goto bar ;
  }
}
Bye.
